UKHomeOffice/nmsw-ui
# NMSW-388 Display Task Details page after General Declaration uploaded

# Ticket

NMSW-388

----

## AC

Given I have just completed the upload general declaration step of the submit voyage process
And I on the 'upload success' page
When I click save and continue
Then I am taken to the task details page

Given I have only uploaded the General declaration (FAL 1)
When I am on the task list page
Then I see my ship name and voyage type displayed

Given I have only uploaded the General declaration (FAL 1)
When I am on the task list page
Then the general declaration has a status of completed
And crew details has a status of required
And passengers has a status of required
And supporting documents has a status of optional
And check your answers has a status of cannot start yet

Given I have only uploaded the General declaration (FAL 1)
When I am on the task list page
Then the link to check your answers is inactive (cannot be clicked)

----

## To test

- Run app
- Sign in
- Click on Report a voyage
- Upload a General declaration (FAL 1) with no errors
- Continue through the success page
- > You should be returned to the task list page
- > You should see the ship name and voyage type (that were in the FAL 1 file you uploaded) displayed
- > The general declaration link should show the status of 'completed'
- > The check your answers link should be inactive (cannot click it)

----

## Notes

This PR does not handle error responses from API endpoint. They will be in next PR
